### PR TITLE
Release 100.8

### PR DESCRIPTION
**Pull request by philium**

### CODE REVIEW

File: arcgis-ios-sdk-samples/Display information/Custom dictionary style/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Custom dictionary style
+
+Use a custom dictionary style (.stylx) to symbolize features using a variety of attribute values.
+
+![Image of custom dictionary style](image1.png)
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Display information/Custom dictionary style/README.md:5]: Should we leave this image not renamed? This sample was created before the README updates started.

File: arcgis-ios-sdk-samples/Geometry/Convex hull/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Convex hull
+
+Create a convex hull for a given set of points. The convex hull is a polygon with shortest perimeter that encloses a set of points. As a visual analogy, consider a set of points as nails in a board. The convex hull of the points would be like a rubber band stretched around the outermost nails.
+
+![Create a convex hull](image1.png)
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Geometry/Convex hull/README.md:5]: Should we leave this image not renamed? This sample was created before the README updates started.

File: arcgis-ios-sdk-samples/Geometry/Nearest vertex/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Nearest vertex
+
+Find the closest vertex and coordinate of a geometry to a point.
+
+![Nearest vertex](image1.png)
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Geometry/Nearest vertex/README.md:5]: Should we leave this image not renamed? This sample was created before the README updates started.

File: arcgis-ios-sdk-samples/Edit data/Create and save KML file/CreateAndSaveKMLViewController.swift [hunk 4]
```diff
@@ -0,0 +1,271 @@
+// Copyright © 2020 Esri.
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Edit data/Create and save KML file/CreateAndSaveKMLViewController.swift:1]: Not a big deal - should there be a trailing period on this line? I saw different versions from different samples. If yes, some samples below need to add it.

File: arcgis-ios-sdk-samples/Layers/Display subtype feature layer/README.metadata.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  "category": "Features",
+  "description": "Display a composite layer of all the subtype values in a feature class.",
+  "ignore": false,
+  "images": [
+    "Display-subtype-feature-laer.png"
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Layers/Display subtype feature layer/README.metadata.json:6]:
```suggestion
    "Display-subtype-feature-layer.png"
```
There is a typo - please double check if my change is correct.

File: arcgis-ios-sdk-samples/Maps/Read a geopackage/README.md [hunk 6]
```diff
@@ -1,14 +1,43 @@
...
-![](image1.png)
-![](image2.png)
+When the sample loads, the feature tables and rasters from the GeoPackage will be shown on the map.
 
 ## How it works
 
-`AGSMap` has a property called `operationalLayers` which is a list of `AGSLayer`. To add a layer the app uses the `add()` method on the list (`AGSList`). To remove a layer it uses the `remove()` method. To re-order layers, the app uses the `remove()` method and the `insert(at:)` method to re-add the layer at the appropriate index.
+1. Initialize the GeoPackage using `AGSGeoPackage`.
+2. Iterate through available rasters, exposed by `AGSGeopackage.geoPackageRasters`.
+    * For each raster, create a new  `AGSRasterLayer`, then add it to the map.
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Maps/Read a geopackage/README.md:19]:
```suggestion
    * For each raster, create a new `AGSRasterLayer`, then add it to the map.
```
Redundant whitespace.

File: arcgis-ios-sdk-samples/Maps/Read a geopackage/README.md [hunk 7]
```diff
@@ -1,14 +1,43 @@
...
+3. Iterate through available feature tables, exposed by `AGSGeopackage.geoPackageFeatureTables`.
+    * For each feature table, create a new  `AGSFeatureLayer`, then add it to the map.
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Maps/Read a geopackage/README.md:21]:
```suggestion
    * For each feature table, create a new `AGSFeatureLayer`, then add it to the map.
```
Redundant whitespace.

File: arcgis-ios-sdk-samples/Maps/Read a geopackage/README.md [hunk 8]
```diff
@@ -1,14 +1,43 @@
...
+* AGSGeoPackageFeatureTable
+* AGSGeoPackageRaster
+
+## Offline data
+
+The [Aurora Colorado GeoPackage](https://www.arcgis.com/home/item.html?id=68ec42517cdd439e81b036210483e8e7) holds datasets that cover Aurora, Colorado.
+
+## About the data
+
+This sample features a GeoPackage with datasets that cover Aurora, Colorado: Public art (points), Bike trails (lines), Subdivisions (polygons), Airport noise (raster), and liquour license density (raster).
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Maps/Read a geopackage/README.md:35]:
```suggestion
This sample features a GeoPackage with datasets that cover Aurora, Colorado: Public art (points), Bike trails (lines), Subdivisions (polygons), Airport noise (raster), and liquor license density (raster).
```
Fix typo.

File: arcgis-ios-sdk-samples/Maps/Read a geopackage/ReadGeopackageViewController.swift [hunk 9]
```diff
@@ -45,32 +45,18 @@ class ReadGeopackageViewController: UIViewController, UIPopoverPresentationContr
...
-    
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        if let gpkgLayersVC = segue.destination as? GPKGLayersViewController {
-            // Provide the map and all layers to the layer controller UI.
-            gpkgLayersVC.map = mapView.map
-            gpkgLayersVC.allLayers = allLayers
-
-            gpkgLayersVC.popoverPresentationController?.delegate = self
+            
+            // Add the arrays of feature and reaster layers to the map.
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Maps/Read a geopackage/ReadGeopackageViewController.swift:49]:
```suggestion
            // Add the arrays of feature and raster layers to the map.
```
Fix typo in comment.

File: arcgis-ios-sdk-samples/Route & Directions/Find closest facility to an incident (interactive)/FindClosestFacilityInteractiveViewController.swift [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+                // Create a point and graphics for the incident.
+                let graphic = AGSGraphic(geometry: mapPoint, symbol: incidentSymbol)
+                self.incidentGraphicsOverlay.graphics.add(graphic)
+                
+                // Set the incident for the parameters.
+                parameters.setIncidents([AGSIncident(point: mapPoint)])
+                self.closestFacilityTask.solveClosestFacility(with: parameters) { [weak self] (result, error) in
+                    guard let self = self else { return }
+                    if let result = result {
+                        // Get the ranked list of colsest facilities.
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Route & Directions/Find closest facility to an incident (interactive)/FindClosestFacilityInteractiveViewController.swift:100]:
```suggestion
                        // Get the ranked list of closest facilities.
```
Fix typo in comment.

File: arcgis-ios-sdk-samples/Route & Directions/Find closest facility to an incident (interactive)/README.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+*   AGSClosestFacilityResult
+*   AGSClosestFacilityRoute
+*   AGSClosestFacilityTask
+*   AGSFacility
+*   AGSGraphic
+*   AGSGraphicsOverlay
+*   AGSIncident
+
+## Tags
+routing & directions, incident, network analysis, route, search
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Route & Directions/Find closest facility to an incident (interactive)/README.md:39]:
```suggestion
## Tags

routing & directions, incident, network analysis, route, search
```
Add an line here.

File: arcgis-ios-sdk-samples/Route & Directions/Find closest facility to an incident (interactive)/README.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+1.  Create a new `AGSClosestFacilityTask` using a URL from an online network analysis service.
+1.  Get `AGSClosestFacilityParameters` from the task.
+1.  Add facilities to the parameters.
+1.  Add an incident (as an `AGSPoint`) to the parameters.
+1.  Get `AGSClosestFacilityResult` by solving the task with the parameters.
+1.  Get the indexed list of closet facilities to the incident.
+1.  Get the index of the closest facility.
+1.  Get closest facility route from the facility result.
+1.  Display the route on the `AGSMapView` as an `AGSGraphic` on an `AGSGraphicsOverlay`.
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Route & Directions/Find closest facility to an incident (interactive)/README.md:25]:
```suggestion
1. Create a new `AGSClosestFacilityTask` using a URL from an online network analysis service.
2. Get `AGSClosestFacilityParameters` from the task.
3. Add facilities to the parameters.
4. Add an incident (as an `AGSPoint`) to the parameters.
5. Get `AGSClosestFacilityResult` by solving the task with the parameters.
6. Get the indexed list of closet facilities to the incident.
7. Get the index of the closest facility.
8. Get closest facility route from the facility result.
9. Display the route on the `AGSMapView` as an `AGSGraphic` on an `AGSGraphicsOverlay`.
```
Fix numbered list.

File: arcgis-ios-sdk-samples/Route & Directions/Navigate route/README.md [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+6. Create an `AGSRouteTracker` using the route result, and the index of the desired route to take.
+7. Use `AGSLocationChangeHandlerDelegate.locationDataSource(_:locationDidChange:)` to track the location of the device and update the route tracking status.
+8. Use `AGSRouteTrackerDelegate.routeTracker(_:didUpdate:)` to be notified of `AGSTrackingStatus` changes, and use them to display updated route information. `AGSTrackingStatus` includes a variety of information on the route progress, such as the remaining distance, remaining geometry or traversed geometry (represented by an `AGSPolyline`), or the remaining time (`Double`), amongst others.
+9. Use `AGSRouteTrackerDelegate.routeTracker(_:didGenerateNewVoiceGuidance:)` to be notified of new voice guidances. From the voice guidance, get the `AGSVoiceGuidance.text` representing the directions and use a text-to-speech engine to output the maneuver directions.
+10. You can also query the tracking status for the current `AGSDirectionManeuver` index, retrieve that maneuver from the `AGSRoute` and get it's direction text to display in the GUI.
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Route & Directions/Navigate route/README.md:26]:
```suggestion
10. You can also query the tracking status for the current `AGSDirectionManeuver` index, retrieve that maneuver from the `AGSRoute` and get its direction text to display in the GUI.
```
Fix typo.

File: arcgis-ios-sdk-samples/Utility Network/Perform valve isolation trace/README.metadata.json [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+        "AGSUtilityTraceFilter",
+        "AGSUtilityTier",
+        "AGSUtilityTraceParameters",
+        "AGSUtilityTraceResult",
+        "AGSUtilityTraceType"
+    ],
+    "snippets": [
+        "PerformValveIsolationTraceViewController.swift"
+    ],
+    "title": "Perform Valve Isolation Trace"
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Utility Network/Perform valve isolation trace/README.metadata.json:50]:
```suggestion
    "title": "Perform valve isolation trace"
```
This should not be all capitalized.

File: arcgis-ios-sdk-samples/Layers/Display subtype feature layer/README.metadata.json [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+  ],
+  "relevant_apis": [
+    "AGSLabelDefinition",
+    "AGSServiceFeatureTable",
+    "AGSSubtypeFeatureLayer",
+    "AGSSubtypeSublayer"
+  ],
+  "snippets": [],
+  "title": "Display subtype feature layer"
+}
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Layers/Display subtype feature layer/README.metadata.json:26]:
```suggestion
}

```
I'd like to have a trailing blank line for any text file.

**Comment by philium** [comment]
@yo1995 Could you please create a PR targeting the release branch with your suggested changes? I wouldn't worry about the period in the copyright comment.

**Comment by yo1995** [comment]
> @yo1995 Could you please create a PR targeting the release branch with your suggested changes? I wouldn't worry about the period in the copyright comment.

Sure. Please refer to #873 .

File: arcgis-ios-sdk-samples/Geometry/Convex hull/ConvexHullViewController.swift [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+            let symbol: AGSSymbol
+            switch convexHullGeometry.geometryType {
+            case .point:
+                symbol = markerSymbol
+            case .polyline:
+                symbol = lineSymbol
+            default:
+                symbol = fillSymbol
+            }
+            // remove the existing graphic for convex hull if there is one
```
Comment (vquach2404) [inline arcgis-ios-sdk-samples/Geometry/Convex hull/ConvexHullViewController.swift:73]:
```suggestion
            // Remove the existing graphic for convex hull if there is one.
```

File: arcgis-ios-sdk-samples/Geometry/Nearest vertex/NearestVertexViewController.swift [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+class NearestVertexViewController: UIViewController {
+    /// The graphics and symbology for the tapped point, the nearest vertex and the nearest coordinate.
+    let tappedLocationSymbol = AGSSimpleMarkerSymbol(style: .X, color: .orange, size: 15)
+    let nearestCoordinateSymbol = AGSSimpleMarkerSymbol(style: .diamond, color: .red, size: 10)
+    let nearestVertexSymbol = AGSSimpleMarkerSymbol(style: .circle, color: .blue, size: 15)
+    
+    /// The symbology for the example polygon area.
+    let polygonFillSymbol = AGSSimpleFillSymbol(style: .forwardDiagonal, color: .green, outline: AGSSimpleLineSymbol(style: .solid, color: .green, width: 2))
+    
+    /// The graphics overlay for the polygon and points..
```
Comment (vquach2404) [inline arcgis-ios-sdk-samples/Geometry/Nearest vertex/NearestVertexViewController.swift:27]:
```suggestion
    /// The graphics overlay for the polygon and points.
```

Comment (philium) [reply arcgis-ios-sdk-samples/Geometry/Nearest vertex/NearestVertexViewController.swift:27]: @vquach2404 Please create a PR with these suggestions.

**Pull request merged**